src/util/semaphore.js

```diff
@@ -25,21 +23,20 @@ class Semaphore {
             this._verbose = Boolean(params && params.verbose);
             if (params.timeout) {
                 this._timeout = params.timeout;
+                this._timeout_error_code = params.timeout_error_code || 'SEMAPHORE_TIMEOUT';
             }
             if (params.work_timeout) {
                 this._work_timeout = params.work_timeout;
+                this._work_timeout_error_code = params.work_timeout_error_code || 'SEMAPHORE_WORKER_TIMEOUT';
             }
-            this._timeout_error_code = params.timeout_error_code || 'SEMAPHORE_TIMEOUT';
-            this._work_timeout_error_code = params.work_timeout_error_code || 'SEMAPHORE_WORKER_TIMEOUT';
-
         }
     }
 
     async _work_cap() {
         return new Promise((resolve, reject) => {
             const err = new Error('Semaphore Worker Timeout');
             err.code = this._work_timeout_error_code;
-            const t = setTimeout(() => reject(err), this._work_timeout || DEFAULT_WORK_TIMEOUT);
+            const t = setTimeout(() => reject(err), this._work_timeout);
```

Comment:
I might be missing it but what happens when there are no `params.work_timeout`?

Reply:
This function won't be called without `_work_cap` with the current code

Reply:
Ok, I can see that 🙂 
Then LGTM.